tools/src/main/java/it/eng/tools/service/ApplicationPropertiesService.java

```diff
@@ -0,0 +1,243 @@
+package it.eng.tools.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import it.eng.tools.exception.ApplicationPropertyErrorException;
+import it.eng.tools.exception.ApplicationPropertyNotFoundAPIException;
+import it.eng.tools.model.ApplicationProperty;
+import it.eng.tools.repository.ApplicationPropertiesRepository;
+import lombok.extern.java.Log;
+
+/**
+ * The PropertiesService class provides methods to interact with properties, including saving, retrieving, and deleting properties.
+ */
+@Service
+@Log
+public class ApplicationPropertiesService {
+
+	private static final String STORED_APPLICATION_PROPERTIES = "storedApplicationProperties";
+
+	private Environment env;
+
+	private final ApplicationPropertiesRepository repository;
+
+	private Sort sortByIdAsc() {
+		return Sort.by("id");
+	}
+
+	public ApplicationPropertiesService(ApplicationPropertiesRepository repository, Environment env) {
+		this.repository = repository;
+		this.env = env;
+	}
+
+	public List<ApplicationProperty> getProperties(String key_prefix) {
+		
+		List<ApplicationProperty> allProperties = null;
+		
+		if(!StringUtils.isBlank(key_prefix)) {
+			allProperties = repository.findByKeyStartingWith(key_prefix, sortByIdAsc());
+		} else {
+			allProperties = repository.findAll(sortByIdAsc());
+		}
+
+		if (allProperties.isEmpty()) {
+			throw new ApplicationPropertyErrorException("Property not found");
+		} else {
+			return allProperties;
+		}
+	}
+
+	public Optional<ApplicationProperty> getPropertyByKey(String key) {
+		Optional<ApplicationProperty> propertyByMongo = repository.findById(key);
+		if(propertyByMongo.isEmpty()) {
+			log.warning(key + " not found in the db, try in application.properties");
+			//Try to keep value from applicatio.properties
+			String propertyValueByApplicationProperty = env.getProperty(key);

```

Comment:
If UI search for property that is not in DB, this logic will insert that property in mongo, and UI user will be able to modify it? What if UI user requests some property that will require connector to restart, like new value for jks/p12 file?
We should not allow UI users, or any user, to have possibility to move properties to Mongo and change value, unless it is intended for property to be in Mongo and UI user can modify its value

Reply:
The UI has to use the Rest API in ApplicationPropertiesAPIController and not directly the methods in the service class.